build.assets/build-test-compat.sh

```diff
@@ -51,25 +51,54 @@ DISTROS=(
 # It will be set to non-zero value if any of run commands returns an error.
 EXIT_CODE=0
 
-# Run binary in a Docker container and propagate returned exit code.
+# Run binary in a Docker container and propagate returned exit code. 
+#
+# This will sometimes run under Google Cloud Build, which implies using Docker-
+# out-of-Docker to interact with containers. This means that simply mounting
+# the test targest into the test container won't work, as it would require
```

Comment:
Are you sure that mounting doesn't work? https://github.com/GoogleCloudPlatform/cloud-builders/issues/372#issuecomment-423692532

Reply:
That works for mounting stuff into the build container, but _not_ for mounting into the nested (well, actually _sibling_) containers we use for testing.

The Docker daemon we're talking to in this script is _outside_ the container that this script is running in. If we try to mount a directory into the child container then path that we give will be interpreted _in the context of that external Docker daemon_ and not our current container. We have no knowledge of how the build container is set up, so we can't translate our local path into one that the Docker daemon will interpret correctly.

If we had more control over the build container creation we could get expose a volume on it and use `docker run --volumes-from` pick it up. Unfortunately the build container is created by GCB and there are no options to _expose_ a volume, only to mount them, and so this is also a non starter.